sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -1747,8 +2116,20 @@ pi_result piextKernelGetNativeHandle(pi_kernel kernel,
 // This API is called by Sycl RT to notify the end of the plugin lifetime.
 // TODO: add a global variable lifetime management code here (see
 // pi_level_zero.cpp for reference) Currently this is just a NOOP.
+// We clear all the 'map' variables here.
```

Comment:
I added the code to clean up all the maps that we are populating in the plugin. I also delete the pi_device handles in the cslice_devices list if they have not been deleted yet. 
Thanks

Reply:
Ah..the code cleanup needs to be done in a different way. I will address this in the next PR. Thanks